Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -0,0 +1,174 @@
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true))
+          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
+
+      var error = "New error just arrived";
+      var resourceManagerStub = new Mock<IResourceManager>();
+      resourceManagerStub.Setup(_ => _.TryGetString("Remotion.ObjectBinding.Web.UI.Controls.BocList.HasValidationFailureText", out error)).Returns(true);
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.GetResourceManager()).Returns(resourceManagerStub.Object);
+      bocListMock.Object.ID = "someBocList";
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ErrorMessage = "This should not be output";
+      validator.ControlToValidate = "someBocList";
+
+      NamingContainer.Controls.Add(bocListMock.Object);
+      NamingContainer.Controls.Add(validator);
+
+      ((IValidatorWithDynamicErrorMessage)validator).RefreshErrorMessage();
+
+      Assert.That(validator.ErrorMessage, Is.EqualTo("New error just arrived"));
+    }
+
+    [Test]
+    public void RefreshErrorMessage_WithBocListValidationFailures_ReturnsFailuresOfBocList ()
+    {
+      var validationFailureRepoStub = new Mock<IBocListValidationFailureRepository>();
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.Object.ID = "someBocList";
+
+      var validationFailuresForBocList = new[]
+                                         {
+                                           BocListValidationFailureWithLocationInformation.CreateFailure(BusinessObjectValidationFailure.Create("Errors on BocList"))
+                                         };
+
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocList(true)).Returns(validationFailuresForBocList);
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true))
+          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ErrorMessage = "This should not be output.";
+      validator.ControlToValidate = "someBocList";
+
+      NamingContainer.Controls.Add(bocListMock.Object);
+      NamingContainer.Controls.Add(validator);
+
+      ((IValidatorWithDynamicErrorMessage)validator).RefreshErrorMessage();
+
+      Assert.That(validator.ErrorMessage, Is.EqualTo("Errors on BocList\r\n"));
+    }
+
+    [Test]
+    public void RefreshErrorMessage_WithInternalValidationFailures_ReturnsLocalisedInfo ()
```

Comment:
`WithInternalValidationFailures`
please explain / better wording

Reply:
internal = BusinessObjectPropertyOfBocList (in my head at least) -> I think WithBocListPropertyValidationFailures should be alright?

Reply:
`WithValidationFailuresForBoundProperty`